### DISCUSSION TITLE
rspamd keeps rejecting spam, even though it should never do that

### DISCUSSION THREAD

**Discussion by Woefdram** [rspamd/rspamd#4658] [General]
My SMTP-servers check incoming messages, and after they're OK, they deliver them to the scanner. Because of this architecture, the scanner can never refuse a message from the SMTP-server, ever. The only thing the scanner can do, is scan it and add a few headers, before passing it on to the IMAP-server.

I've configured "reject = null" in actions.conf, that should be enough for what I read everywhere, but every message containing the GTUBE-string is still being rejected. I understand that this string was specifically devised to trigger a reject, but if I specifically configure rspamd to not reject messages, it shouldn't reject it, right?

Its current behaviour makes it impossible for me to test and evaluate the scoring, before I move rspamd to production. How do I convince rspamd to never EVER reject a message from my SMTP-servers

I've already configured them as external_relays, so rspamd should already be aware of them.

**Comment by fatalbanana** [comment]
GTUBE is always rejected; that's unfortunately not subject to any configuration.

You might use some message designed to enrage Rspamd for testing:

~~~
From: foo
From: foo
Date: bar
Date: bar
X-KLMS-AntiSpam-Status: spam
X-Spam: yes
X-Forefront-Antispam-Report: SFV:SPM
X-UI-Filterresults: junk
Content-type: text/html

<a href="http://example.net/hell&#8203;o">hello</a>
~~~

**Reply by Woefdram** [reply]
Nice tip, does exactly what I want. For other people looking into this stuff, here's how you add those headers in Thunderbird:

https://www.lifewire.com/arbitrary-custom-heading-email-thunderbird-1173089

**Comment by Woefdram** [comment]
Thanks @fatalbanana. Not the anwer I was hoping for, but if that's what it is, I'll have to deal with it.

**Comment by moisseev** [comment]
> to test and evaluate the scoring

You could temporarily enable and use the `add header` pattern [1] instead of GTUBE, but it won't help with `scoring` as Rspamd triggers the action unconditionally.

1. [GTUBE-like test patterns](https://rspamd.com/doc/gtube_patterns.html)

**Reply by Woefdram** [reply]
Yes, I've tried all four of them already, and they do exactly what their description says. Only, I had expected I could override that behaviour in my config.
